### PR TITLE
[webui][api] port to ruby 2.4

### PR DESCRIPTION
**Pull request by adrianschroeter**
Just get it working, do not care about deprecated warnings yet

### CODE REVIEW

**Comment by adrianschroeter** [comment]
Lots of deprecated warnings, so I am not sure if we really want to merge this already.....

**Comment by adrianschroeter** [comment]
rubygems we want to update to avoid deprecate warnings: 

rails
concurrent-ruby
sprockets
builder
nokogiri
riddle
concurrent-rub
ruby_parser

**Comment by adrianschroeter** [comment]
Ana is working on gem updates independent of this, so I dropped the gem updates here again.

**Comment by hennevogel** [comment]
@adrianschroeter please rebase

**Comment by hennevogel** [comment]
Hm this stupid github conflict resolve added a commit. @adrianschroeter please rebase manually and get rid  of it. TIA

File: src/api/test/unit/attrib_test.rb [hunk 1]
```diff
@@ -48,12 +48,13 @@ class AttribTest < ActiveSupport::TestCase
...
     attrib = Attrib.new(attrib_type: attrib_type, package: Package.first)
 
     attrib_type.value_count = 1
     attrib.values << attrib_value
     assert attrib.valid?, "attrib should be valid: #{attrib.errors.messages}"
-    attrib.values << attrib_value
+    attrib.values << attrib_value_second
     assert_not attrib.valid?
     assert_equal ["has 2 values, but only 1 are allowed"], attrib.errors.messages[:values]
     attrib.values.delete_all
```
Comment (bgeuken) [inline src/api/test/unit/attrib_test.rb]: Looking at the commit message ([ci] fixing testcase with new rails activerecord 5.0.1). Is it in this PR intentionally or by accident?

**Comment by bgeuken** [comment]
LGTM,

If we merge this now, this would mean that next release (OBS 2.8) would run on ruby.2.4. Are you fine with that @openSUSE/open-build-service ?

**Comment by hennevogel** [comment]
I'm fine with it, we can still move back if the beta phase shows that this is not stable enough...

File: src/api/test/unit/attrib_test.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/test/unit/attrib_test.rb]: half and half ... it was anyway a correct bugfix, it was just not discovered before 5.0.1, so I kept it (after removing the gem updates commit again from this pull request).

**Comment by hennevogel** [comment]
Let's go then.

**Pull request merged**